office365_connector.py

```diff
@@ -569,17 +569,22 @@ def _make_rest_call(self, action_result, url, verify=True, headers={}, params=No
         except AttributeError:
             return RetVal(action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)), resp_json)
 
-        try:
-            r = request_func(
-                url,
-                data=data,
-                headers=headers,
-                verify=verify,
-                params=params,
-                timeout=MSGOFFICE365_DEFAULT_REQUEST_TIMEOUT)
-        except Exception as e:
-            error_msg = _get_error_message_from_exception(e)
-            return RetVal(action_result.set_status(phantom.APP_ERROR, "Error connecting to server. {0}".format(error_msg)), resp_json)
+        for _ in range(self._number_of_retries):
+            try:
+                r = request_func(
+                                url,
+                                data=data,
+                                headers=headers,
+                                verify=verify,
+                                params=params,
+                                timeout=MSGOFFICE365_DEFAULT_REQUEST_TIMEOUT)
+            except Exception as e:
+                error_msg = _get_error_message_from_exception(e)
+                return RetVal(action_result.set_status(phantom.APP_ERROR, "Error connecting to server. {0}".format(error_msg)), resp_json)
+            if r.status_code != 502:
+                break
```

Comment:
should we add debug_log on the response here?

(e.g. `self.debug_print('Received response: %s' % r)` )

Reply:
Debugging this way would print the response object and not the content. We have included a debug statement "Received 502 status code from the server" and also regardless of the response content, with 502 status code we have added the retry mechanism